src/bin/packageJson.ts

```diff
@@ -0,0 +1,16 @@
+import { readFileSafeAsJson } from "../shared/readFileSafeAsJson.js";
+
+interface PackageWithVersion {
+	version?: string;
+}
+
+export async function getVersionFromPackageJson(): Promise<string | undefined> {
+	const path = new URL("../../package.json", import.meta.url);
+	const data = (await readFileSafeAsJson(path)) as PackageWithVersion;
+
+	if (typeof data === "object" && typeof data.version === "string") {
+		return data.version;
+	}
+
+	return undefined;
```

Comment:
[Cleanup] Hmm, in what situation would `data` legitimately not be an object with `"version"`? I think this actually would be a reasonable place to throw an error. That way the `bin` function won't need to check if it's defined.

```ts
const data = ... as Partial<PackageWithVersion> | undefined;
```

What do you think?